### ISSUE TITLE
User Interface modifications for grouped layers

### ISSUE THREAD

**Issue by dan-bowerman** [fgpv-vpgf/fgpv-vpgf#303]
- [x] Define legend schema for group layers (#334)
- [x] Wire TOC to new configuration schema

Traceability Matrix ID#106

**Comment by dan-bowerman** [comment]
Need to define legend schemas first? See separate PBI #334.

**Comment by mweech** [comment]
Blocked on #334

**Comment by AleksueiR** [comment]
Hit some problems as `geoService` got a touch messy and splitting layer-related code into a separate service caused circular dependencies. Had to refactor. See PR #381 for details.

**Comment by AleksueiR** [comment]
See update in #394.

**Comment by AleksueiR** [comment]
Can't yet generate symbology for wms and esriImage layers. Develop demo page updated.

**Comment by AleksueiR** [comment]
Linking layer/sublayer visibility with legend turned out to be harder than I thought.

**Comment by AleksueiR** [comment]
Merged into develop.
